### DISCUSSION TITLE
Library Panels "equivalent" for queries.

### DISCUSSION THREAD

**Discussion by Kayak007** [grafana/grafana#59881] [Dashboards, Panels & Visualizations]
Hi,

We have a lot of KPIs (based on a Prometheus Data Source) with formulas that keep evolving.
Some of these KPIs are used in various "Official" Dashboards/Panels that we modify each time the formula evolves.
Which is very time consuming.
Furthermore, these "Official" Dashboards/Panels are sometimes copied by other users for their own needs, and those are not updated with the formula change. Leading to several problems...

The need would be to have an equivalent to the Library Panels feature, but for queries. A kind of KPI repository.
It would associate a KPI name with a query (and a Data Source?) and several tags.
This KPI name could in turn be used in a panel instead of the query formula.
And each time we modify the formula in the KPI repository it would be taken into account in the Dashboard/Panels where it's used.

If there's an existing solution, please share.

Thanks.

Regards.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
